src/components/interaction-designer/block-editors/choices/mixins/CommonVoiceChoiceConfig.vue

```diff
@@ -0,0 +1,81 @@
+<script>
+import {mapActions} from 'vuex'
+import {IChoice, ISelectOneResponseBlock} from '@floip/flow-runner'
+import {BLOCK_TYPE} from '@/store/flow/block-types/MobilePrimitives_SelectOneResponseBlockStore'
+
+export default {
+  props: {
+    block: {
+      type: ISelectOneResponseBlock,
+      required: true,
+    },
+    choice: {
+      type: IChoice,
+      required: true,
+    },
+    index: {
+      type: Number,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      options: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '0', '*'],
+      blockResponseExpression: 'block.response',
+    }
+  },
+  computed: {
+    currentExpressionWithoutSpaces() {
+      return this.currentExpression !== undefined ? this.currentExpression.replace(/\s/g, '') : undefined
+    },
+    currentExpression() {
+      return this.choice.ivr_test?.test_expression
+    },
+    /**
+     * Test if:
+     * - the expression starts with `block.response='` (floip schema) or `@block.response='` (from partner's flow)
+     * - and if the estimated key_press is part of the options' elements
+     */
+    isKeyPressIdentifiable() {
+      if (this.currentExpressionWithoutSpaces?.startsWith(`${this.blockResponseExpression}='`) === true
+        || this.currentExpressionWithoutSpaces?.startsWith(`@${this.blockResponseExpression}='`) === true) {
+        const estimatedKeyPress = this.currentExpressionWithoutSpaces.split('\'')?.[1]
```

Comment:
maybe use `estimatedKeyPress` from computed values?

Reply:
I've tried that before and got a TS error
<img width="1082" alt="Screen Shot 2022-08-04 at 12 32 21 PM" src="https://user-images.githubusercontent.com/68745918/182928752-6e52b49f-5cf9-4e0d-a0fb-a51c22ce031f.png">

I think I'm going to use it and ignore the TS error for now (I've google a while but couldn't find the solution :D ).

Reply:
Interesting. Yes, I'm not sure either (it seems like we need to do `bind(this)` or use an arrow function). I'm okay with leaving this as it is 👍